### PR TITLE
refactor PublicShareDialogFragment

### PR DESCRIPTION
**Pull request by theScrabi**
This also fixed a bug regarding the link password edit. There was a button with which you can toggle the password visibility. That button was not visible before due to a bug. Now it is.

### CODE REVIEW

File: src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 1]
```diff
@@ -104,17 +99,37 @@
...
+    private View mEditPermissionSelectionView;
+    private View mShowFileListingSelectionView;
+    private SwitchCompat mEditPermissionSwitch;
+    private SwitchCompat mShowFileListingSwitch;
+    private TextView mPasswordLabel;
+    private SwitchCompat mPasswordSwitch;
+    private EditText mPasswordValueEdit;
+    private TextView mExpirationDateLabel;
+    private SwitchCompat mExpirationDateSwitch;
+    private TextView mExpirationDateExplenationLabel;
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: There's a typo here, mExpirationDateExpl**a**nationLabel?

File: src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 2]
```diff
@@ -197,105 +213,83 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-                boolean publicLinkEditPermissions = getEditPermissionSwitch(getView()).isChecked();
-
-                // Show file listing option
-                boolean publicLinkSupportOnlyUpload = !getShowFileListingSwitch(getView()).
+                final String publicLinkName = mNameValueEdit.getText().toString();
+                String publicLinkPassword = mPasswordValueEdit.getText().toString();
+                final long publicLinkExpirationDateInMillis = getExpirationDateValueInMillis();
+                boolean publicLinkEditPermissions = mEditPermissionSwitch.isChecked();
+                final boolean publicLinkSupportOnlyUpload = !mShowFileListingSwitch.
                         isChecked();
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: This fits properly in the previous line

File: src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java [hunk 3]
```diff
@@ -560,13 +531,10 @@ public void onDetach() {
...
-     * @param shareView Root view in the fragment.
      */
-    private void initAllowEditingListener(View shareView) {
-        mOnAllowEditingInteractionListener = new OnAllowEditingInteractionListener();
-
-        ((SwitchCompat) shareView.findViewById(R.id.shareViaLinkEditPermissionSwitch)).
-                setOnCheckedChangeListener(mOnAllowEditingInteractionListener);
+    private void initAllowEditingListener() {
+        mEditPermissionSwitch.setOnCheckedChangeListener(
+                new OnAllowEditingInteractionListener());
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/PublicShareDialogFragment.java]: It fits properly in the previous line as well

**Comment by davigonz** [comment]
@theScrabi just a couple of minor changes, good refactoring, thanks for getting rid of all those getters. 

giphy 3

**Comment by davigonz** [comment]
Code approved, ready to test @jesmrec

**Comment by jesmrec** [comment]
Checks performed over the public share fragment:

- [x] Link name: create and edit
- [x] Allow edit: create and edit
- [x] Show file list: create and edit
- [x] Password: create and edit. Force and unforce (capability)
- [x] Expiration: create and edit. Force and unforce (capability)

**Pull request merged**